pkg/operator/ceph/cluster/crash/crash.go

```diff
@@ -123,56 +127,112 @@ func (r *ReconcileNode) createOrUpdateCephCrash(node corev1.Node, tolerations []
 }
 
 // createOrUpdateCephCron is a wrapper around controllerutil.CreateOrUpdate
-func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, cephVersion *version.CephVersion) (controllerutil.OperationResult, error) {
-	cronJob := &v1beta1.CronJob{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      prunerName,
-			Namespace: cephCluster.GetNamespace(),
-		},
-	}
-	err := controllerutil.SetControllerReference(&cephCluster, cronJob, r.scheme)
-	if err != nil {
-		return controllerutil.OperationResultNone, errors.Errorf("failed to set owner reference of deployment %q", cronJob.Name)
-	}
+func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, cephVersion *cephver.CephVersion, k8sVersion *version.Version) (controllerutil.OperationResult, error) {
+	minVersionForCronV1 := "1.21.0"
 
-	// Adding volumes to pods containing data needed to connect to the ceph cluster.
-	volumes := controller.DaemonVolumesBase(config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath), "")
-	volumes = append(volumes, keyring.Volume().CrashCollector())
+	// minimum k8s version required for v1 cronJob is 'v1.21.0'. check if k8s version is atleast 'v1.21.0' apply V1 or else v1beta1 cronJob.
+	if k8sVersion.AtLeast(version.MustParseSemantic(minVersionForCronV1)) {
+		cronJob := &v1.CronJob{
```

Comment:
I think the only repetitive code should be which cronJob version to use - v1 or v1beta1. Rest of the code can be reused. I would suggest creating a new function that compares the k8s version and return appropriate cronJob. And have unit test for this method.

Reply:
I'll add unit test for `createOrUpdateCephCron`

Reply:
added.